### DISCUSSION TITLE
how can i launch this workflow-extension in langium 3.0.0

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1415] [Q&A] [answered]
i intergate langium 3.0.0 with glsp(refer to glsp workflow example) ,since langium 3.0.0 is ESM module(i used to use langium 1.1.0,which is commonjs  ,vscode extension can launch well),when i press f5 and lanunch vscode extension ,an error occured,so how can i launch this workflow-extension
![image](https://github.com/eclipse-langium/langium/assets/26995040/389f7194-10ba-44ba-a9ea-8995a3873e80)
![image](https://github.com/eclipse-langium/langium/assets/26995040/46325cec-a46b-482f-af38-104ec2fda6dc)

**Comment by zrwsmd** [comment]
i have change extension module  webpack.js to ESM module ,so output filename is  .mjs suffix
![image](https://github.com/eclipse-langium/langium/assets/26995040/b3f5bf26-370b-493b-b55d-4bab19da7d8d)

**Comment by zrwsmd** [comment]
whole project structure like below screenshot
![image](https://github.com/eclipse-langium/langium/assets/26995040/6a6e3d64-90f6-4521-bb43-d96137976923)

**Comment by msujew** [comment, marked as answer]
@zrwsmd you need to bundle your code to cjs to be compatible with vscode. See also [our own extension bundling code](https://github.com/eclipse-langium/langium/blob/f6889f23b6462dd388dfc9f1e787a44bd29a7c1d/packages/langium-vscode/esbuild.mjs#L37).
